oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/OneandoneComputeServiceAdapter.java

```diff
@@ -0,0 +1,288 @@
+      if (ram < 1024) {
+         ram = 0.5;
+      } else {
+         ram = ram / 1024;
+      }
+
+      try {
+         List<? extends Processor> processors = hardware.getProcessors();
+         org.apache.jclouds.oneandone.rest.domain.Hardware.CreateHardware hardwareRequest
+                 = org.apache.jclouds.oneandone.rest.domain.Hardware.CreateHardware.create(cores, 1, ram, hdds);
+         final Server.CreateServer serverRequest = Server.CreateServer.builder()
+                 .name(name)
+                 .description(name)
+                 .hardware(hardwareRequest)
+                 .rsaKey(options.getPublicKey())
+                 .password(privateKey == null ? password : null)
+                 .applianceId(image.getId())
+                 .dataCenterId(dataCenterId)
+                 .powerOn(Boolean.TRUE).build();
+
+         logger.trace("<< provisioning server '%s'", serverRequest);
+
+         server = api.serverApi().create(serverRequest);
+
+         waitServerUntilAvailable.apply(server);
+
+         updateServer = api.serverApi().get(server.id());
+
+         Map<Integer, Integer> portsRange = getPortRangesFromList(inboundPorts);
+         List<FirewallPolicy.Rule.CreatePayload> rules = new ArrayList<FirewallPolicy.Rule.CreatePayload>();
+
+         for (Map.Entry<Integer, Integer> range : portsRange.entrySet()) {
+            FirewallPolicy.Rule.CreatePayload rule = FirewallPolicy.Rule.CreatePayload.builder()
+                    .portFrom(range.getKey())
+                    .portTo(range.getValue())
+                    .protocol(Types.RuleProtocol.TCP)
+                    .build();
+            rules.add(rule);
+         }
+         if (inboundPorts.length > 0) {
+            FirewallPolicy rule = api.firewallPolicyApi().create(FirewallPolicy.CreateFirewallPolicy.create(server.name() + " firewall policy", "desc", rules));
+            api.serverApi().addFirewallPolicy(updateServer.id(), updateServer.ips().get(0).id(), rule.id());
+            waitServerUntilAvailable.apply(server);
+         }
+
+         logger.trace(">> provisioning complete for server. returned id='%s'", server.id());
+
+      } catch (Exception ex) {
+         logger.error(ex, ">> failed to provision server. rollbacking..");
+         throw Throwables.propagate(ex);
+      }
+
+      LoginCredentials serverCredentials = LoginCredentials.builder()
+              .user(loginUser)
+              .password(password)
+              .privateKey(privateKey)
+              .build();
+
+      return new NodeAndInitialCredentials<Server>(updateServer, updateServer.id(), serverCredentials);
+   }
+
+   @Override
+   public List<HardwareFlavour> listHardwareProfiles() {
+      return api.serverApi().listHardwareFlavours();
+   }
+
+   @Override
+   public Iterable<SingleServerAppliance> listImages() {
+      GenericQueryOptions options = new GenericQueryOptions();
+      options.options(0, 0, null, Types.OSImageType.Minimal.toString(), null);
```

Comment:
Is is a problem to list them all? If it is, we could add a context property for a string to configure this search, so users can customize, if they want to filter images, the type of images they're interested in.